### DISCUSSION TITLE
warning: Unmapped target property: "chunks"

### DISCUSSION THREAD

**Discussion by Skjolberg** [mapstruct/mapstruct#3371] [Q&A]
Hello :)

When I do the mapping, it shows me: warning: **Unmapped target property: "chunks"**

I don't have any attribute named "chunks", the getChunks() method has no relation to any attribute named like that. I have noticed that whatever I put it tries to map it, but if I remove the "get" and put for example only "chunks", the warning does not appear. 

The temporary solution I have is to use: @ Mapping(target = "chunks", ignore = true)

As annotations that I am using in the class are from lombok: 
- @ Data
- @ AllArgsConstructor
- @ NoArgsConstructor

Is this normal? Is it a Lombok thing? Thx

```java
public List<Chunk> getChunks() {
        List<Chunk> res = new ArrayList<>();

        World w = getWorld();
        int x1 = getLowerX() & ~0xf;
        int x2 = getUpperX() & ~0xf;
        int z1 = getLowerZ() & ~0xf;
        int z2 = getUpperZ() & ~0xf;
        for (int x = x1; x <= x2; x += 16) {
            for (int z = z1; z <= z2; z += 16) {
                res.add(w.getChunkAt(x >> 4, z >> 4));
            }
        }
        return res;
}

**Comment by filiphr** [comment]
MapStruct does not know anything about your class fields. MapStruct is using the JavaBean convention to determine properties that should be mapped.

In this particular case it treats `chunks` as a property since there is `getChunks`. Since `chunks` is a `List` we treat the getter as a setter. I think that we have an issue for this. However, I can't find it in the moment
